app/models/spree/user_decorator.rb

```diff
@@ -4,7 +4,7 @@
   devise :omniauthable
 
   def apply_omniauth(omniauth)
-    if %w(facebook google_oauth2).include? omniauth['provider']
+    if omniauth.fetch('info', {})['email'].present?
```

Comment:
TIL!
It looks like dig was introduced in Ruby 2.3. gemspec requires >= 1.9.3, so we could use it and add a polyfill?
let me know what you'd like to do